lib/mime-info/filename.js

```diff
@@ -0,0 +1,263 @@
+    * value-chars   = *( pct-encoded / attr-char )
+    * pct-encoded   = "%" HEXDIG HEXDIG
+                   ; see [RFC3986], Section 2.1
+    * attr-char     = ALPHA / DIGIT
+                   / "!" / "#" / "$" / "&" / "+" / "-" / "."
+                   / "^" / "_" / "`" / "|" / "~"
+                   ; token except ( "*" / "'" / "%" )
+    *
+    * egHeader: attachment;filename*=utf-8''%E4%BD%A0%E5%A5%BD.txt
+    */
+    // eslint-disable-next-line max-len
+    encodedFileNameRegex = /;[ \t]*(?:filename\*)[ \t]*=[ \t]*([A-Za-z0-9!#$%&+\-^_`{}~]+)'.*'((?:%[0-9A-Fa-f]{2}|[A-Za-z0-9!#$&+.^_`|~-])+)[ \t]*/,
+
+    /**
+    * @private
+    *
+    * RegExp to match quoted-pair in RFC 2616
+    *
+    * quoted-pair = "\" CHAR
+    * CHAR        = <any US-ASCII character (octets 0 - 127)>
+    *
+    */
+    quotedPairRegex = /\\([ -~])/g,
+
+    /**
+     * @private
+     * Regex to match file base name and extension name
+     * Group 1 will be matched to the base name
+     * Group 2 will be matched to the extension name
+     */
+    fileNameSplitRegex = /^(.+?\.?)(?:\.([^.]*$)|$)/,
+
+    /**
+     * @private
+     * Regex to match all the hexadecimal number inside encoded string
+     */
+    hexCharMatchRegex = /%([0-9A-Fa-f]{2})/g,
+
+    /**
+     * @private
+     * Regex to match non-latin characters
+     */
+    nonLatinCharMatchRegex = /[^\x20-\x7e\xa0-\xff]/g,
+
+    /**
+     * Replaces non-latin characters with '?'
+     *
+     * @private
+     * @param {string} val - The string which needs to be checked
+     */
+    getLatin = function (val) {
+        return String(val).replace(nonLatinCharMatchRegex, '?');
+    },
+
+    /**
+     * Decodes the hexcode to charCode
+     *
+     * @private
+     * @param {string} str
+     * @param {string} hex
+     * @returns {string} String with decoded hexcode values
+     */
+    decodeHexcode = function (str, hex) {
+        return String.fromCharCode(parseInt(hex, 16));
+    },
+
+    /**
+     * HashMap for decoding string with supported characterSet
+     *
+     * @private
+     */
+    characterDecodeHashMap = {
+        [characterSetEnum.iso88591]: function (encodedString) {
+            return getLatin(encodedString);
+        },
+        [characterSetEnum.utf8]: function (encodedString) {
+            if (!supportsBuffer) {
+                return;
+            }
+            return Buffer.from(encodedString, BINARY).toString(characterSetEnum.UTF8);
+        }
+    },
+
+    /**
+     * Decodes the given filename with given charset
+     * The supported charset are iso-8859-1 and utf-8
+     *
+     * @private
+     * @param {string} encodedFileName
+     * @param {string} charset
+     * @returns {string} Returns the decoded filename
+     */
+    decodeFileName = function (encodedFileName, charset) {
+        if (!encodedFileName) {
+            return;
+        }
+        if (typeof characterDecodeHashMap[charset] !== 'function') {
```

Comment:
we can check for `!characterDecodeHashMap[charset]`. This is our internal map, we know it will be a function if it is truthy,

Reply:
ok will update